src/Commands/pm/SecurityUpdateCommands.php

```diff
@@ -84,6 +85,40 @@ public function suggestComposerCommand($updates)
         $this->logger()->notice("If that fails due to a conflict then you must update one or more root dependencies.");
     }
 
+    /**
+     * Return file_get_contents context if proxy is detected.
+     *
+     * Drupal settings override ENV ones.
+     *
+     * @return resource
+     *  A stream context resource.
+     */
+    protected function getProxySettings()
```

Comment:
It might make sense to move this somewhere more central, so other commands can benefit.

Reply:
Yes, as I said in the issue they are more than 20 uses of  `file_get_contents ` in drush, it should  be a good idea to replace them with a helper method that manage proxy. But it's a bit out of my league...